File: backend/core-service/src/main/java/com/zzz/core/api/dto/UserRegisterRequest.java
package com.zzz.core.api.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserRegisterRequest {
    @Email
    @NotBlank
    private String email;

    @NotBlank
    private String password;

    @NotBlank
    private String nickname;
}

File: _native_backup/android/app/src/main/java/com/zzz/android/data/remote/dto/UserRegisterRequest.kt
package com.zzz.android.data.remote.dto

import com.google.gson.annotations.SerializedName

data class UserRegisterRequest(
    @SerializedName("email") val email: String,
    @SerializedName("password") val password: String,
    @SerializedName("nickname") val nickname: String
)
